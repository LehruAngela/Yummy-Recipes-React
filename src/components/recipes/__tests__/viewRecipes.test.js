import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ViewRecipe from '../viewRecipes';

const mockStore = configureStore();

let store;
let wrapper;
const initialState = {
  recipe: {
    recipes: [],
    total: 0
  }
};

describe('Navbar Component', () => {
  beforeEach(() => {
    store = mockStore(initialState)
    wrapper = shallow(<ViewRecipe store={store} />)
  });

  it('renders without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
