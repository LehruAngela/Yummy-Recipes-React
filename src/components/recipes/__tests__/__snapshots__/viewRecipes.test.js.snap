// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders without crashing 1`] = `
<ContextProvider
  value={null}
>
  <ViewRecipe
    deleteRecipe={[Function]}
    editRecipe={[Function]}
    pageChange={[Function]}
    recipes={Array []}
    searchRecipe={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    total={0}
    viewRecipe={[Function]}
  />
</ContextProvider>
`;
